kythe/rust/indexer/src/indexer/analyzers.rs

```diff
@@ -413,16 +416,9 @@ impl<'a, 'b> CrateAnalyzer<'a, 'b> {
             }
 
             if let Some(krate_id) = self.krate_ids.get(&def.id.krate) {
-                let krate_signature = format!(
-                    "{}_{}_{}",
-                    krate_id.disambiguator.0, krate_id.disambiguator.1, krate_id.name
-                );
-
                 // Generate node based on definition type
                 let mut def_vname = self.krate_vname.clone();
-                let def_signature = format!("{}_def_{}", krate_signature, def.id.index);
-                def_vname.set_signature(def_signature.clone());
-                def_vname.set_language("rust".to_string());
+                def_vname.set_signature(self.create_def_signature(krate_id, def.id.index));
                 def_vname.clear_path();
```

Comment:
The krate_vname (from which this was cloned) shouldn't have a path. I get this is defensive. Maybe instead of `create_def_signature` have a `create_def_v_name(base_vname, krate_id, index)` ?